src/Tezos/Address.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Address.h"
+#include "BinaryCoding.h"
+#include "Hash.h"
+#include "HexCoding.h"
+#include "../Data.h"
+
+#include <TrezorCrypto/base58.h>
+#include <TrezorCrypto/ecdsa.h>
+#include <TrustWalletCore/TWBase58.h>
+#include <TrezorCrypto/base58.h>
+
+using namespace TW;
+using namespace TW::Tezos;
+
+bool Address::isValid(const std::string& string) {
+    size_t capacity = 128;
+    uint8_t buffer[capacity];
+    int size = base58_decode_check(string.data(), HASHER_SHA2D, buffer, (int)capacity);
+
+    // verify prefix
+    std::array<std::uint8_t, 3> prefix {6, 161, 159};
+    for (size_t i = 0; i< prefix.size(); i++) {
+        uint8_t byte = buffer[i];
+        if (prefix[i] != byte) {
+            return false;
+        }
+    }
+    return size == Address::size;
+}
+
+Address::Address(const std::vector<uint8_t>& data) {
+    assert(data.size() == size);
```

Comment:
Assert only works in debug mode, this will potentially cause segfault in release. See https://github.com/TrustWallet/wallet-core/pull/100

Reply:
Done.